Code/Tools/AssetProcessor/native/unittests/RCcontrollerUnitTests.cpp

```diff
@@ -736,42 +801,36 @@ void RCcontrollerUnitTests::RunRCControllerTests()
     jobdetailsC.m_jobDependencyList.push_back({ jobDependencyD });
     jobdetailsD.m_jobDependencyList.push_back({ jobDependencyC });
 
-    MockRCJob* jobD = new MockRCJob(&m_rcController.m_RCJobListModel);
-    MockRCJob* jobC = new MockRCJob(&m_rcController.m_RCJobListModel);
+    MockRCJob* jobD = new MockRCJob(m_rcJobListModel);
+    MockRCJob* jobC = new MockRCJob(m_rcJobListModel);
 
     jobC->Init(jobdetailsC);
-    m_rcController.m_RCQueueSortModel.AddJobIdEntry(jobC);
-    m_rcController.m_RCJobListModel.addNewJob(jobC);
+    m_rcQueueSortModel->AddJobIdEntry(jobC);
+    m_rcJobListModel->addNewJob(jobC);
 
     jobD->Init(jobdetailsD);
-    m_rcController.m_RCQueueSortModel.AddJobIdEntry(jobD);
-    m_rcController.m_RCJobListModel.addNewJob(jobD);
+    m_rcQueueSortModel->AddJobIdEntry(jobD);
+    m_rcJobListModel->addNewJob(jobD);
 
-    m_rcController.SetDispatchPaused(false);
-    m_rcController.DispatchJobs();
-    UNIT_TEST_EXPECT_TRUE(UnitTestUtils::BlockUntil(allJobsCompleted, MaxProcessingWaitTimeMs));
+    m_rcController->SetDispatchPaused(false);
+    m_rcController->DispatchJobs();
+    EXPECT_TRUE(UnitTestUtils::BlockUntil(allJobsCompleted, MaxProcessingWaitTimeMs));
 
     // Test case when source file is deleted before it started processing
     {
-        int prevJobCount = rcJobListModel->itemCount();
+        int prevJobCount = m_rcJobListModel->itemCount();
         MockRCJob rcJobAddAndDelete;
         AssetProcessor::JobDetails jobDetailsToInitWithInsideScope;
         jobDetailsToInitWithInsideScope.m_jobEntry.m_pathRelativeToWatchFolder = jobDetailsToInitWithInsideScope.m_jobEntry.m_databaseSourceName = "someFile0.txt";
         jobDetailsToInitWithInsideScope.m_jobEntry.m_platformInfo = { "pc",{ "tools", "editor" } };
         jobDetailsToInitWithInsideScope.m_jobEntry.m_jobKey = "Text files";
-        jobDetailsToInitWithInsideScope.m_jobEntry.m_sourceFileUUID = jobDetailsToInitWith.m_jobEntry.m_sourceFileUUID;
+        jobDetailsToInitWithInsideScope.m_jobEntry.m_sourceFileUUID = AZ::Uuid("{D013122E-CF2C-4534-A87D-F82570FBC2CD}");
         rcJobAddAndDelete.Init(jobDetailsToInitWithInsideScope);
-        rcJobListModel->addNewJob(&rcJobAddAndDelete);
+        m_rcJobListModel->addNewJob(&rcJobAddAndDelete);
         // verify that job was added
```

Comment:
nit: Use line breaks or some visual clue to separate the GIVEN WHEN THEN blocks in your tests

Reply:
I'm mostly migrating the existing test code instead of rewriting them or adding new tests. As @AMZN-stankowi pointed out, some of these legacy tests may not make sense anymore and require to rewrite. Maybe we can address all these in another story.